### DISCUSSION TITLE
16 bit/channel RGBCW

### DISCUSSION THREAD

**Discussion by Warx44** [Makuna/NeoPixelBus#767] [Q&A] [answered]
How hard will be to implement a Rgb80Color::Max? same timing with WS2812. the chips are used in indoor/outdoor lights and have 5 channels of 16bits RGBCW. The chips I am trying to control are SM16705

**Comment by Makuna** [comment]
(ask questions in the discussion area)

There is already a Rgbww80Color, and it has a Max.  There is not a NeoRgbcw80Feature yet, but it is trivial to create.

The SM16705 is not what you want.  Are you sure you have the right chip number?  That one is basically a shift register.  It takes a single 8 bit value clocked in, then a latch signal is triggered by the uc in which it will then switch the 8 outputs on or off based on that 8 bit value.  This is not the type of chips that NeoPixelBus works with.

**Reply by Warx44** [reply]
thanks for the quick response , the only IC is a SM16705 , I did get the leds to work with a RGBCW 80bit stream , but I need to make them work from Wled using NeoPixelBus. I will try to make a NeoRgbcw80Feature.

**Reply by Makuna** [reply, marked as answer]
I just checked in NeoRgbww80Features (both Rgbcw and Rgbwc).

Do you have a link to the spec sheet.  When I look up SM16705, I get what I described.

**Reply by Warx44** [reply]
datasheet: https://www.linkage.cn/uploads/allimg/20231204/2-2312041G613933.pdf
Not in English but you can get the idea , same timings with ws2812 except 5 ch 16bit

**Comment by Warx44** [comment]
Got it to work , I made a rgbcw80 feature using 'Rgbww80Color' and using the 'Neo800KbpsMethod' it works. 
The lights are here: https://www.amazon.com/gp/product/B0C85M2KLS/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&th=1.
Now that I can send the right data next step is to make it work with Wled and 8bit colors.

**Comment by Makuna** [comment]
Eight bit element color like RgbwwColor will auto convert and can be used when calling SetPixelColor().  WLED I believe is only eight bit element colors and they rely on this.  They already have support for a Rgbw64Color feature I believe.
